crates/consensus/authority/src/block_builder.rs

```diff
@@ -181,22 +185,41 @@ where
         };
         drop(storage);
 
-        // Process Botanix specific logs and get current block pegouts
-        let current_block_pegouts = match crate::utils::process_receipts(
-            &mut self.btc_server.clone(),
-            &bundle_state,
-            botanix_consensus_pkg.bitcoin_checkpoint.1,
-            self.btc_network,
-            self.consensus.chain_spec.parent_confirmation_depth,
-        )
-        .await
-        {
-            Ok(pegouts) => pegouts,
-            Err(e) => {
-                error!(target: "consensus::authority", ?e, "Failed to process botanix log");
-                return;
+        // Process pegins and pegouts from the [Minting] contract.
+        let mut block_pegouts = Vec::new();
+        for (idx, receipts) in bundle_state.receipts().iter().enumerate() {
+            for receipt in receipts {
+                if idx == 0 && receipt.is_none() {
+                    break; // Prunning block, skip
+                }
+                if let Some(receipt) = receipt {
+                    if !receipt.success {
+                        continue;
+                    }
+                    for log in &receipt.logs {
+                        let pegin_match = try_parse_mint_event(log).expect("passed EVM check");
```

Comment:
Comment justifiying the expect
```suggestion
                        // Mint event should have already been validated in during evm execution (in processor.rs) 
                        let pegin_match = try_parse_mint_event(log).expect("passed EVM check");
```